src/Internal/Service/MinimumAmountService.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * MinimumAmountService class.
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Service;
+
+use WC_Payments_Utils;
+use WCPay\Internal\Proxy\LegacyProxy;
+
+/**
+ * Minimum Amount Service to handle errors from the server when the payment amount is too small.
+ */
+class MinimumAmountService {
+	/**
+	 * Transient key prefix.
+	 */
+	private const TRANSIENT_KEY = 'wcpay_minimum_amount_';
+
+	/**
+	 * Legacy Proxy.
+	 *
+	 * @var LegacyProxy
+	 */
+	private $legacy_proxy;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param LegacyProxy $legacy_proxy Legacy proxy instance.
+	 */
+	public function __construct( LegacyProxy $legacy_proxy ) {
+		$this->legacy_proxy = $legacy_proxy;
+	}
+
+	/**
+	 * Saves the minimum amount required for transactions in a given currency.
+	 *
+	 * @param string $currency The currency.
+	 * @param int    $amount   The minimum amount in cents.
+	 */
+	public function set_cache( string $currency, int $amount ): void {
```

Comment:
I'd call this `set_cached_amount`, and `get_cache` could be `get_cached_amount`. WDYT?